src/ir_Funiki.cpp

```diff
@@ -0,0 +1,580 @@
+        break;
+      default:
+        new_position = kFunikiSwingLastPos;
+    }
+  } else {
+      new_position = kFunikiSwingAuto;
+  }
+  _.SwingV = new_position;
+}
+
+/// Get the Vertical Swing Automatic mode setting of the A/C.
+/// @return true, the setting is on. false, the setting is off.
+bool IRFunikiAC::getSwingVerticalAuto(void) const {
+  return (_.SwingV == kFunikiSwingAuto);
+}
+
+/// Get the Vertical Swing position setting of the A/C.
+/// @return The native position/mode.
+uint8_t IRFunikiAC::getSwingVerticalPosition(void) const { return _.SwingV; }
+
+
+
+/// Get the clock time of the A/C unit.
+/// @return Nr. of seconds past midnight.
+int16_t IRFunikiAC::getClock(void) const {
+  return (bcdToUint8((_.Hours1 << kNibbleSize) |  _.Hours2) * 60)
+  + bcdToUint8((_.Minutes1 << kNibbleSize) |  _.Minutes2);
+}
+
+
+void IRFunikiAC::setClock(const int16_t nr_of_minutes) {
+  uint32_t remaining = nr_of_minutes;
+  _.Minutes1 = uint8ToBcd(remaining % 60)>>4;
+  _.Minutes2 = uint8ToBcd(remaining % 60) & 0x0F;
+  remaining /= 60;
+  _.Hours1 = uint8ToBcd(remaining % 60)>>4;
+  _.Hours2 = uint8ToBcd(remaining % 60) & 0x0F;
+}
+
+
+// /// Set the timer enable setting of the A/C.
+// /// @param[in] on true, the setting is on. false, the setting is off.
+// void IRFunikiAC::setTimerEnabled(const bool on) { _.TimerEnabled = on; }
+
+// /// Get the timer enabled setting of the A/C.
+// /// @return true, the setting is on. false, the setting is off.
+bool IRFunikiAC::getTimerOnEnabled(void) const { return _.TimerOnEnable; }
+bool IRFunikiAC::getTimerOffEnabled(void) const { return _.TimerOffEnable; }
+
+// /// Get the timer time value from the A/C.
+// /// @return The number of minutes the timer is set for.
+uint16_t IRFunikiAC::getTimerOn(void) const {
+  uint16_t hrs = irutils::bcdToUint8((_.TimerOnHours1 << kNibbleSize) |
+    _.TimerOnHours2);
+  return hrs * 60 + (_.TimerOnMinutes * 10);
+}
+// /// Get the timer time value from the A/C.
+// /// @return The number of minutes the timer is set for.
+uint16_t IRFunikiAC::getTimerOff(void) const {
+  uint16_t hrs = irutils::bcdToUint8((_.TimerOffHours1 << kNibbleSize) |
+    _.TimerOffHours2);
+  return hrs * 60 + (_.TimerOffMinutes * 10);
+}
+
+// /// Set the A/C's timer to turn off in X many minutes.
+// /// @param[in] minutes The number of minutes the timer should be set for.
+// /// @note Stores time internally in 30 min units.
+// /// e.g. 5 mins means 0 (& Off), 95 mins is  90 mins (& On). Max is 24 hours.
+// void IRFunikiAC::setTimer(const uint16_t minutes) {
+//   uint16_t mins = std::min(kFunikiTimerMax, minutes);  // Bounds check.
+//   setTimerEnabled(mins >= 30);  // Timer is enabled when >= 30 mins.
+//   uint8_t hours = mins / 60;
+//   // Set the half hour bit.
+//   _.TimerHalfHr = (mins % 60) >= 30;
+//   // Set the "tens" digit of hours.
+//   _.TimerTensHr = hours / 10;
+//   // Set the "units" digit of hours.
+//   _.TimerHours = hours % 10;
+// }
```

Comment:
Please remove the commented out function. Try to avoid including code that doesn't work etc. Someone might think it works and try it.